### DISCUSSION TITLE
Config File for CPD?

### DISCUSSION THREAD

**Discussion by cschroedl-gov** [pmd/pmd#3790] [Q&A]
PMD can be configured via standalone `ruleset.xml` files. Is there an equivalent for CPD?

**Comment by adangel** [comment]
No, there is only little to configure for CPD. All what is configurable is available as command-line arguments: https://pmd.github.io/latest/pmd_userdocs_cpd.html
with `--minimum-tokens` the most important option to control how small/large the detected duplications should be.
Especially there is no equivalent of rules like in PMD.

**Comment by oowekyala** [comment]
As @adangel said, there is no XML config file for CPD. However, if you want to use the CLI with fewer parameters, you can put arguments into a dedicated file using [`@` syntax](https://jcommander.org/#_syntax). Put your arguments in a file, one per line, then call CPD with the argument `@path/to/argfile`.

**Reply by adangel** [reply]
are you sure, this works with our old jcommander version? If so, we should probably document it....

**Reply by oowekyala** [reply]
I've never tested it for cpd but I use it all the time for pmd, and they use the same jcommander. We should also add a test for it :)

**Reply by adangel** [reply]
Didn't know this :) Then it should work indeed
